se2_planning_ros/include/se2_planning_ros/GridMapLazyStateValidatorRos.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * GridMapLazyStateValidatorRos.hpp
+ *
+ *  Created on: Jul 20, 2020
+ *      Author: meyerc
+ */
+
+#pragma once
+
+#include <ros/ros.h>
+
+#include <geometry_msgs/Point32.h>
+#include <geometry_msgs/Polygon.h>
+#include <boost/thread/locks.hpp>
+#include <boost/thread/shared_mutex.hpp>
+#include <grid_map_ros/grid_map_ros.hpp>
+#include "se2_planning/GridMapLazyStateValidator.hpp"
+#include "se2_planning_ros/OmplReedsSheppPlannerRos.hpp"
+
+#include <string>
+
+namespace se2_planning {
+
+using WriteLock = boost::unique_lock<boost::shared_mutex>;
+using ReadLock = boost::shared_lock<boost::shared_mutex>;
+
+struct GridMapLazyStateValidatorRosParameters {
+  std::string gridMapFrame_ = "map";
+  std::string gridMapMsgSubTopic_ = "state_validator_ros/traversability_map_in";
+  std::string gridMapMsgPubTopic_ = "state_validator_ros/traversability_map_out";
+  std::string gridMapObstacleLayerName_ = "traversability";  // redundant, already defined by GridMapStateValidator
+  StateValidityCheckingMethod gridMapStateValidityCheckingMethod_ =
+      StateValidityCheckingMethod::TRAVERSABILITY;  // redundant, already defined by GridMapStateValidator
+  double gridMapStateValidityThreshold_ = 0.5;
+  double gridMapUnsafeStateValidityThreshold_ = 0.5;
+  int gridMapMaxNumberOfUnsafeCells_ = 0;
+  double gridMapResolution_ = 0.2;
```

Comment:
why do we need these parameters (resolution, map position etc.)? GridMapLazyStateValidator will receive a map anyway over ROS if I understand correctly. Should that be enough to initialize them?

Reply:
I have to think about it again. The original idea was that it is possible to define the resolution independent of the provided map (downsample it if needed, does not have to be in the node, could happen outside). The rest is probably just there because it was convenient in the beginning.

Reply:
Ok, finally tried that out. The issue is that the GridMapLazyStateValidator precomputes footprint points in its initialize function (GridMapLazyStateValidator::initialize()). This requires that a map is set and uses it geometry to calculate the footprint points. I think that was also the reason why I changed the validation function. 
Could you explain  what the point is behind the computeFootprintPoints function? Maybe we can set it up differently to avoid initializing some random map.
With the current implementation of isInCollision and isTraversable, I get the strange behavior that along the x-axis the state validator accepts all possible positions, along the y-axis only the ones that lie in the map.